File: music-album-webapp/Entities/Album.cs
namespace music_album_webapp.Entities;

public class Album
{
    public int Id { get; set; }
    public string Title { get; set; }
    public string Author { get; set; }
    public string Version { get; set; }
    public int ReleaseYear { get; set; } 
    
    public int DistributionId { get; set; }
    public virtual Distribution Distribution { get; set; }
    
    public virtual List<Track> Tracks { get; set; }
}

File: music-album-webapp/ClientApp/src/app/models/Album.ts
import {Track} from "./Track";

export class Album {

  constructor(id: number, title: string, author: string, version: string | null, releaseYear: number | null, distributionName: string | null, tracks: Array<Track> | null) {
    this.id = id;
    this.title = title;
    this.author = author;
    this.version = version;
    this.releaseYear = releaseYear;
    this.distributionName = distributionName;
    this.tracks = tracks;
  }

  id: number;
  title: string;
  author: string;
  version: string | null;
  releaseYear: number | null;
  distributionName: string | null;
  tracks: Array<Track> | null;
}
